sbncode/SBNEventWeight/App/SystToolsEventWeight_module.cc

```diff
@@ -0,0 +1,238 @@
+////////////////////////////////////////////////////////////////////////
+// Class: SystToolsEventWeight
+// Module Type: producer
+// File: SystToolsEventWeight_module.cc
+//
+// Generated at Mon Feb 21 09:36:11 2015 by Jaesung Kim (jae.sung.kim.3426@gmail.com)
+//
+////////////////////////////////////////////////////////////////////////
+
+#include <iostream>
+#include <assert.h>
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Persistency/Common/PtrMaker.h"
+#include "canvas/Persistency/Common/Assns.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+
+#include "sbnobj/Common/SBNEventWeight/EventWeightMap.h"
+#include "sbnobj/Common/SBNEventWeight/EventWeightParameterSet.h"
+
+#include "systematicstools/interface/ISystProviderTool.hh"
+#include "systematicstools/interpreters/ParamHeaderHelper.hh"
+#include "systematicstools/utility/ParameterAndProviderConfigurationUtility.hh"
+#include "systematicstools/utility/exceptions.hh"
+
+#include "canvas/Persistency/Common/Assns.h"
+#include "art/Framework/Principal/Run.h"
+
+namespace sbn {
+  namespace evwgh {
+
+class SystToolsEventWeight : public art::EDProducer {
+public:
+  explicit SystToolsEventWeight(fhicl::ParameterSet const& p);
+
+  SystToolsEventWeight(SystToolsEventWeight const &) = delete;
+  SystToolsEventWeight(SystToolsEventWeight &&) = delete;
+  SystToolsEventWeight& operator = (SystToolsEventWeight const&) = delete;
+  SystToolsEventWeight& operator = (SystToolsEventWeight&&) = delete;
+
+private:
+  void produce(art::Event& e) override;
+  void beginRun(art::Run& run) override;
+
+private:
+  std::string fGenieModuleLabel;
+  bool fAllowMissingTruth;
+  bool fDebugMode;
+
+  systtools::provider_list_t fSystProviders;
+  systtools::param_header_map_t fParamHeaderMap;
+  systtools::ParamHeaderHelper fParamHeaderHelper;
+
+};
+
+
+SystToolsEventWeight::SystToolsEventWeight(fhicl::ParameterSet const& p)
+  : EDProducer{p}
+{
+
+  fGenieModuleLabel = p.get<std::string>("generator_module_label", "generator");
+  fAllowMissingTruth = p.get<bool>("AllowMissingTruth");
+  fDebugMode = p.get<bool>("debugMode");
```

Comment:
Please be consistent with the parameter name conventions.
You managed to use three different styles.
My recommendation is to use CamelCase:
```suggestion
  fGenieModuleLabel = p.get<std::string>("GeneratorModuleLabel", "generator");
  fAllowMissingTruth = p.get<bool>("AllowMissingTruth");
  fDebugMode = p.get<bool>("DebugMode", false);
```
Also, add a default value for `DebugMode`, since supposedly most people won't neither want to debug nor to bother to say so.
Consider also if it makes sense to have a default value for `AllowMissingTruth` — I suggest `false`, which will force the users to think whether the unusual case where there is no missing truth is expected or not.

Reply:
Updated as suggested. As we changed the fcl parameter name, I will follow up with another update in `icarucode` where we have fcl file for this module.